core/src/main/scala/akka/persistence/r2dbc/internal/codec/TimestampCodec.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2022 - 2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.r2dbc.internal.codec
+
+import java.time.Instant
+import java.time.LocalDateTime
+import java.util.TimeZone
+
+import io.r2dbc.spi.Row
+import io.r2dbc.spi.Statement
+
+import akka.annotation.InternalApi
+import akka.persistence.r2dbc.internal.InstantFactory
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] sealed trait TimestampCodec {
+
+  def encode(timestamp: Instant): Any
+  def decode(row: Row, name: String): Instant
+  def decode(row: Row, index: Int): Instant
+
+  // should we name it just `now()`? The type should not be in the name...
+  def instantNow(): Instant = InstantFactory.now()
```

Comment:
I don't think we should have this at all in the TimestampCodec. It's not used by the rich statement and we can use `InstantFactory.now()` explicitly at the places where this is currently being used. The indirection via codec doesn't only adds confusion for creating an Instant.